buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -306,9 +321,33 @@ public void plugin(String pluginProjectPath) {
     @Override
     public void module(Provider<RegularFile> module) {
         checkFrozen();
+        registerExtractedConfig(module);
         this.modules.add(module.map(RegularFile::getAsFile));
     }
 
+    private void registerExtractedConfig(Provider<RegularFile> pluginProvider) {
+        if (bundleAttribute == null) {
+            bundleAttribute = Attribute.of("bundle", Boolean.class);
+            project.getDependencies().getAttributesSchema().attribute(bundleAttribute);
+            project.getDependencies().getArtifactTypes().maybeCreate(ArtifactTypeDefinition.ZIP_TYPE);
+            project.getDependencies().registerTransform(UnzipTransform.class, transformSpec -> {
+                transformSpec.getFrom()
+                    .attribute(ArtifactAttributes.ARTIFACT_FORMAT, ArtifactTypeDefinition.ZIP_TYPE)
+                    .attribute(bundleAttribute, true);
+                transformSpec.getTo()
+                    .attribute(ArtifactAttributes.ARTIFACT_FORMAT, ArtifactTypeDefinition.DIRECTORY_TYPE)
+                    .attribute(bundleAttribute, true);
+                transformSpec.getParameters().setAsFiletreeOutput(true);
+            });
+        }
+        Dependency pluginDependency = this.project.getDependencies().create(project.files(pluginProvider));
+        Configuration extractedConfig = project.getConfigurations().detachedConfiguration(pluginDependency);
+        extractedConfig.getAttributes().attribute(ArtifactAttributes.ARTIFACT_FORMAT, ArtifactTypeDefinition.DIRECTORY_TYPE);
+        extractedConfig.getAttributes().attribute(bundleAttribute, true);
+        pluginAndModuleConfiguration.builtBy(pluginProvider);
+        pluginAndModuleConfiguration.from(extractedConfig);
```

Comment:
Is this bit necessary? Does creating a dependency via `project.files(RegularFile)` not retain task dependency information?

Reply:
fixed